### PR TITLE
Vehicle structure

### LINKED ISSUE #8: Vehicle reference coordinate system
Original author: @ClemensLinnhoff 

In [model_structure](https://github.com/LudwigFriedmann/OpenMaterial/tree/master/model_structure), the reference coordinate system of a vehicle is placed on the front axle with x facing backwards.

To my knowledge, the x-direction contradicts DIN ISO 8855, where it is specified as facing forward. Most of the defined sub-components also have the x-axis facing forward.
In OSI the reference origin is typically located at the rear axle [(see fig. 10)](https://opensimulationinterface.github.io/osi-documentation/#_coordinate_systems_and_reference_points). This coincides with the definition of the [sensor coordinate frame](https://github.com/LudwigFriedmann/OpenMaterial/tree/master/model_structure#338-sensors) here.
Why is the vehicle reference frame not also located at the rear axle with the x-axis facing forward? Wouldn't that simplify coordinate transformations?

### PR DESCRIPTION
**Pull request by FKlopfer**
## Describe your changes

- Added the adoc files for the chapter about the vehicle structure in the geometry subgroup.
- There is an index files that order the adoc files.
- Every adoc file contains some default content based on the legacy material.

fixes #8, #10

### CODE REVIEW

**Comment by FKlopfer** [comment]
@mettelilienthal , please update the files.

File: Grp_Vehicle_Part_keyword_vehicle_part_idx (T).adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+| *z-axis*
+| ...
+|===
+
+
+Add a figure.
+
+.<Figure caption>
+//image::images/Vehicle_Structure_Door_Coord_Frame.svg[width=70%, scalewidth=10cm]
```
Comment (mettelilienthal) [inline Grp_Vehicle_Part_keyword_vehicle_part_idx (T).adoc:28]:
```suggestion
//image::images/Vehicle_Structure_Door_Coord_Frame.svg[width=70%, scalewidth=10cm]
////The structure can be extended by custom groups for additional use-cases. Custom groups shall follow the general structure (e.g. grouping into static/dynamic, interior/exterior).  ////
```

File: content/geometry/vehicle-structure/Grp_Interior_Dynamic.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+| *z-axis*
+| ...
+|===
+
+
+Add a figure.
+
+.<Figure caption>
+//image::images/Vehicle_Structure_Door_Coord_Frame.svg[width=70%, scalewidth=10cm]
```
Comment (mettelilienthal) [inline content/geometry/vehicle-structure/Grp_Interior_Dynamic.adoc:28]:
```suggestion
//image::images/Vehicle_Structure_Door_Coord_Frame.svg[width=70%, scalewidth=10cm]

////
Comment SG: Any need for moving pedals or shift levers, or the levers on the steering wheel? 
////
```

File: content/geometry/geometry-index.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
+= Vehicle structure
+:toc:
+:toclevels: 4
+
+include::vehicle-structure/Grp_Vehicle_Part_keyword_vehicle_part_idx.adoc[leveloffset=+1]
+include::vehicle-structure/Grp_Exterior.adoc[leveloffset=+2]
+include::vehicle-structure/Grp_Exterior_Dynamic.adoc[leveloffset=+3]
```
Comment (LudwigFriedmann) [inline content/geometry/geometry-index.adoc:7]: Reference to Grp_Exterior_Static is missing.

**Review by LudwigFriedmann — CHANGES_REQUESTED** [review]

File: content/geometry/geometry-index.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
+= Vehicle structure
+:toc:
+:toclevels: 4
+
+include::vehicle-structure/Grp_Vehicle_Part_keyword_vehicle_part_idx.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/geometry-index.adoc]: File was inserted in the wrong folder. Should work now

File: content/geometry/geometry-index.adoc (hunk 3 above)
Comment (FKlopfer) [reply content/geometry/geometry-index.adoc:8]: Reference added.

**Review by LudwigFriedmann — CHANGES_REQUESTED** [review]

File: content/geometry/vehicle-structure/Grp_Door_Right_door_right_idx_(T).adoc [hunk 5]
```diff
@@ -0,0 +1,28 @@
+= Grp_Door_Top_<door_top_idx> (T)
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/vehicle-structure/Grp_Door_Right_door_right_idx_(T).adoc]: Insert correct content

File: content/geometry/geometry-index.adoc [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+include::vehicle-structure/Grp_Light_Tail_Lefttail_left_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Tail_Righttail_right_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Brake_Leftbrake_left_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Brake_Centerbrake_center_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Brake_Rightbrake_right_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Reverse_Leftreverse_left_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Reverse_Rightreverse_right_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Fog_Leftfog_left_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Fog_Rightfog_right_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Corner_Leftcorner_left_idx_(T).adoc[leveloffset=+3]
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/geometry-index.adoc:37]: Add missing link

File: content/geometry/vehicle-structure/Grp_Side_Mirror_View_Left side_mirror_view_left_idx_(T).adoc [hunk 7]
```diff
@@ -0,0 +1,28 @@
+= Grp_Side_Mirror_Mounting_Left <side_mirror_mounting_left_idx> (T)
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/vehicle-structure/Grp_Side_Mirror_View_Left side_mirror_view_left_idx_(T).adoc]: Insert correct content

File: content/geometry/geometry-index.adoc [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+include::vehicle-structure/Grp_Light_Day_Rightday_right_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Indicator_Left_indicator_left_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Indicator_Right_indicator_right_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Number_Plate_number_plate_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Number_Plate_number_plate_light_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Light_Warningwarning_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Convertible_Top.adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Sensors_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Side_Mirror_Mounting_Left side_mirror_mounting_left_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Side_Mirror_View_Left side_mirror_view_left_idx_(T).adoc[leveloffset=+3]
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/geometry-index.adoc]: Change order

File: content/geometry/geometry-index.adoc [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+include::vehicle-structure/Grp_Blindspot_Mirror_Mounting blindspot_mirror_mounting_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Blindspot_Mirror_View blindspot_mirror_view_idx_(T).adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Interior.adoc[leveloffset=+2]
+include::vehicle-structure/Grp_Interior_Static.adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Interior_Dynamic.adoc[leveloffset=+3]
+include::vehicle-structure/Grp_Steering_Wheel_(T).adoc[leveloffset=+4]
+include::vehicle-structure/Grp_Eyepoint_(T).adoc[leveloffset=+4]
+include::vehicle-structure/Grp_Rearview_Mirror_Mounting rearview_mirror_mounting_idx_(T).adoc[leveloffset=+4]
+include::vehicle-structure/Grp_Rearview_Mirror_View rearview_mirror_view_idx_(T).adoc[leveloffset=+4]
+
```
[the remark this answers is not in the document]
Comment (FKlopfer) [reply content/geometry/geometry-index.adoc:60]: Add "notes" file

**Review by LudwigFriedmann — COMMENTED** [review]
In order to integrate the model structure in an overarching document, I'd extend the table of content by the chapters "Introduction", "General" and "Model Creation":

|- Introduction
|- General
|- Model Creation
|- Model Structure
|    |- ... current toc content ...
|- Notes

**Comment by FKlopfer** [comment]
Updated file structure and add new topics.

File: content/geometry/geometry-index.adoc [hunk 10]
```diff
@@ -0,0 +1,65 @@
+= Vehicle structure
+:toc:
+:toclevels: 4
+
+
+include::vehicle-structure/introduction.adoc[leveloffset=+1]
```
Comment (ClemensLinnhoff) [inline content/geometry/geometry-index.adoc:6]: In my opinion it is a bit counter-intuitive to add individual elements of a certain object structure to the general geometry index. I would have expected to find this index in the vehicle-structure folder for the elements of the vehicle structure. Is that possible?

Comment (ClemensLinnhoff) [inline content/geometry/geometry-index.adoc:6]: Do we actually need individual files for all the elements? What is the benefit compared to putting all element descriptions in one file "vehicle-structure.adoc"?

Comment (ClemensLinnhoff) [reply content/geometry/geometry-index.adoc:6]: This also contradicts the general structure of the Geometry chapter introduced in #36, in my opinion.

**Comment by ClemensLinnhoff** [comment]
This PR should fix #10, so I added it to the PR description. Or does anyone disagree?

File: content/geometry/geometry-index.adoc (hunk 10 above)
Comment (FKlopfer) [reply content/geometry/geometry-index.adoc:6]: Individual files for all elements have the following advantages:
- Information on a certain topic are easier to find in the file structure
- Group members can work on different topics at the same time without interference and possible merge conflicts 
- Changes can be tracked more easily
- Topics can be moved more easily in the file structure or even between chapters
- Linking to individual topics is easier in AsciiDoc than linking to subsections 

We will have to align on how the chapter structure of the document and of the subchapters (e.g. geometry) looks like.

Comment (FKlopfer) [reply content/geometry/geometry-index.adoc:6]: Group consensus: But topics in one file

**Comment by ClemensLinnhoff** [comment]
@LudwigFriedmann could you re-review the changes made by @FKlopfer based on your remarks?
I would suggest we make this structural MR ready to be merged and fill in the descriptions and definitions in a new MR.

**Comment by LudwigFriedmann** [comment]
The table of content seems to be removed from this chapter. As it serves as visualization of the hierarchical structure, can we have it back?

**Comment by ClemensLinnhoff** [comment]
@LudwigFriedmann and I just noticed, that the vehicle structure overview was remove in one of the commits, probably while switching from multiple files to just one file.

Please add the structure overview again as the first subsection under "Model structure":

```
|-------Grp_Vehicle_Part_<keyword>_<vehicle_part_idx> (T)
| |-------Grp_Exterior
| |  |-------Grp_Exterior_Static
| |  |-------Grp_Exterior_Dynamic
| |  |-------Grp_Wheel_<axle_idx>_<wheel_idx> (T)
| |  |  |-------Grp_Wheel_<axle_idx>_<wheel_idx>_Steering
| |  |  |-------Grp_Wheel_<axle_idx>_<wheel_idx>_Steering_Rotating (T)
| |  |-------Grp_Door_Front_<door_front_idx> (T)
| |  |-------Grp_Door_Left_<door_left_idx> (T)
| |  |-------Grp_Door_Right_<door_right_idx> (T)
| |  |-------Grp_Door_Rear_<door_rear_idx> (T)
| |  |-------Grp_Door_Top_<door_top_idx> (T)
| |  |-------Grp_Door_Bottom_<door_bottom_idx> (T)
| |     |-------Grp_Light_Day_Left<day_left_idx> (T)
| |     |-------Grp_Light_Day_Right<day_right_idx> (T)
| |  |-------Grp_Light_Low_Beam_Left<low_beam_left_idx> (T)
| |     |-------Grp_Light_Low_Beam_Right<low_beam_right_idx> (T)
| |  |-------Grp_Light_High_Beam_Left<high_beam_left_idx> (T)
| |     |-------Grp_Light_High_Beam_Right<high_beam_right_idx> (T)
| |  |-------Grp_Light_Position_Left<position_left_idx> (T)
| |     |-------Grp_Light_Position_Right<position_right_idx> (T)
| |     |-------Grp_Light_Park_Left<position_left_idx> (T)
| |     |-------Grp_Light_Park_Right<position_right_idx> (T)
| |  |-------Grp_Light_Tail_Left<tail_left_idx> (T)
| |     |-------Grp_Light_Tail_Right<tail_right_idx> (T)
| |  |-------Grp_Light_Brake_Left<brake_left_idx> (T)
| |     |-------Grp_Light_Brake_Center<brake_center_idx> (T)
| |     |-------Grp_Light_Brake_Right<brake_right_idx> (T)
| |  |-------Grp_Light_Reverse_Left<reverse_left_idx> (T)
| |     |-------Grp_Light_Reverse_Right<reverse_right_idx> (T)
| |  |-------Grp_Light_Fog_Left<fog_left_idx> (T)
| |     |-------Grp_Light_Fog_Right<fog_right_idx> (T)
| |     |-------Grp_Light_Corner_Left<corner_left_idx> (T)
| |     |-------Grp_Light_Corner_Right<corner_right_idx> (T)
| |  |-------Grp_Light_Indicator_Left_<indicator_left_idx> (T)
| |  |-------Grp_Light_Indicator_Right_<indicator_right_idx> (T)
| |  |-------Grp_Number_Plate_<number_plate_idx> (T)
| |  |-------Grp_Light_Number_Plate_<number_plate_light_idx> (T)
| |     |-------Grp_Light_Warning<warning_idx> (T)
| |  |-------Grp_Convertible_Top
| |  |-------Grp_Sensors (T)
| |  |-------Grp_Side_Mirror_Mounting_Left <side_mirror_mounting_left_idx> (T)
| |  |-------Grp_Side_Mirror_View_Left <side_mirror_view_left_idx> (T)
| |  |-------Grp_Side_Mirror_Mounting_Right <side_mirror_mounting_right_idx> (T)
| |  |-------Grp_Side_Mirror_View_Right <side_mirror_view_right_idx> (T)
| |     |-------Grp_Blindspot_Mirror_Mounting <blindspot_mirror_mounting_idx> (T)
| |     |-------Grp_Blindspot_Mirror_View <blindspot_mirror_view_idx> (T)
| |-------Grp_Interior
| |  |-------Grp_Interior_Static
| |  |-------Grp_Interior_Dynamic
| |  | |-------Grp_Steering_Wheel (T)
| |  | |-------Grp_Eyepoint (T)
| |  | |-------Grp_Rearview_Mirror_Mounting <rearview_mirror_mounting_idx> (T)
| |  | |-------Grp_Rearview_Mirror_View <rearview_mirror_view_idx> (T)
```

The individual elements of this hierarchy should be linked to the respective sections.

Please also check, if we can make this hierarchy look a bit nicer. [Here](https://nikostotz.de/blog/tree-in-asciidoctor.html) is a page describing different solutions. I personally like the described solution with using Creole Markup via PlantUML very much. @nakohdo could you implement this, since @FKlopfer is on vacation?

**Comment by ClemensLinnhoff** [comment]
I talked to @nakohdo. @mettelilienthal will fix the hierarchy overview next week.

**Comment by ClemensLinnhoff** [comment]
> I talked to @nakohdo. @mettelilienthal will fix the hierarchy overview next week.

What is the status here @mettelilienthal ?

**Comment by mettelilienthal** [comment]
Both Fabian and I have researched methods of showing hierarchy levels in AsciiDoc. While the described solution with using Creole Markup via PlantUML seems promising, we have not been able to test the final publication view so far. Frank is working on the publication pipeline and we hope to have it completed by the end of the week.
We therefore suggest adding the hierarchy levels in their basic form, but refrain from adding the section links until we know how the hierarchy levels will be displayed. Does that work for you?

**Comment by ClemensLinnhoff** [comment]
Okay so you suggest du add them just as a list and then later come back to make them look nicer? I would be fine with that.

File: content/geometry/vehicle-structure-model-structure.adoc [hunk 11]
```diff
@@ -0,0 +1,1569 @@
+= Model structure
+
```
Comment (mettelilienthal) [inline content/geometry/vehicle-structure-model-structure.adoc:2]:
```suggestion

== Table of content
|-------Grp_Vehicle_Part_<keyword>_<vehicle_part_idx> (T)
| |-------Grp_Exterior
| |  |-------Grp_Exterior_Static
| |  |-------Grp_Exterior_Dynamic
| |  |-------Grp_Wheel_<axle_idx>_<wheel_idx> (T)
| |  |  |-------Grp_Wheel_<axle_idx>_<wheel_idx>_Steering
| |  |  |-------Grp_Wheel_<axle_idx>_<wheel_idx>_Steering_Rotating (T)
| |  |-------Grp_Door_Front_<door_front_idx> (T)
| |  |-------Grp_Door_Left_<door_left_idx> (T)
| |  |-------Grp_Door_Right_<door_right_idx> (T)
| |  |-------Grp_Door_Rear_<door_rear_idx> (T)
| |  |-------Grp_Door_Top_<door_top_idx> (T)
| |  |-------Grp_Door_Bottom_<door_bottom_idx> (T)
| |     |-------Grp_Light_Day_Left<day_left_idx> (T)
| |     |-------Grp_Light_Day_Right<day_right_idx> (T)
| |  |-------Grp_Light_Low_Beam_Left<low_beam_left_idx> (T)
| |     |-------Grp_Light_Low_Beam_Right<low_beam_right_idx> (T)
| |  |-------Grp_Light_High_Beam_Left<high_beam_left_idx> (T)
| |     |-------Grp_Light_High_Beam_Right<high_beam_right_idx> (T)
| |  |-------Grp_Light_Position_Left<position_left_idx> (T)
| |     |-------Grp_Light_Position_Right<position_right_idx> (T)
| |     |-------Grp_Light_Park_Left<position_left_idx> (T)
| |     |-------Grp_Light_Park_Right<position_right_idx> (T)
| |  |-------Grp_Light_Tail_Left<tail_left_idx> (T)
| |     |-------Grp_Light_Tail_Right<tail_right_idx> (T)
| |  |-------Grp_Light_Brake_Left<brake_left_idx> (T)
| |     |-------Grp_Light_Brake_Center<brake_center_idx> (T)
| |     |-------Grp_Light_Brake_Right<brake_right_idx> (T)
| |  |-------Grp_Light_Reverse_Left<reverse_left_idx> (T)
| |     |-------Grp_Light_Reverse_Right<reverse_right_idx> (T)
| |  |-------Grp_Light_Fog_Left<fog_left_idx> (T)
| |     |-------Grp_Light_Fog_Right<fog_right_idx> (T)
| |     |-------Grp_Light_Corner_Left<corner_left_idx> (T)
| |     |-------Grp_Light_Corner_Right<corner_right_idx> (T)
| |  |-------Grp_Light_Indicator_Left_<indicator_left_idx> (T)
| |  |-------Grp_Light_Indicator_Right_<indicator_right_idx> (T)
| |  |-------Grp_Number_Plate_<number_plate_idx> (T)
| |  |-------Grp_Light_Number_Plate_<number_plate_light_idx> (T)
| |     |-------Grp_Light_Warning<warning_idx> (T)
| |  |-------Grp_Convertible_Top
| |  |-------Grp_Sensors (T)
| |  |-------Grp_Side_Mirror_Mounting_Left <side_mirror_mounting_left_idx> (T)
| |  |-------Grp_Side_Mirror_View_Left <side_mirror_view_left_idx> (T)
| |  |-------Grp_Side_Mirror_Mounting_Right <side_mirror_mounting_right_idx> (T)
| |  |-------Grp_Side_Mirror_View_Right <side_mirror_view_right_idx> (T)
| |     |-------Grp_Blindspot_Mirror_Mounting <blindspot_mirror_mounting_idx> (T)
| |     |-------Grp_Blindspot_Mirror_View <blindspot_mirror_view_idx> (T)
| |-------Grp_Interior
| |  |-------Grp_Interior_Static
| |  |-------Grp_Interior_Dynamic
| |  | |-------Grp_Steering_Wheel (T)
| |  | |-------Grp_Eyepoint (T)
| |  | |-------Grp_Rearview_Mirror_Mounting <rearview_mirror_mounting_idx> (T)
| |  | |-------Grp_Rearview_Mirror_View <rearview_mirror_view_idx> (T)
```

File: content/geometry/vehicle-structure-model-structure.adoc [hunk 12]
```diff
@@ -0,0 +1,1624 @@
...
+| |  |-------Grp_Side_Mirror_View_Right <side_mirror_view_right_idx> (T)
+| |     |-------Grp_Blindspot_Mirror_Mounting <blindspot_mirror_mounting_idx> (T)
+| |     |-------Grp_Blindspot_Mirror_View <blindspot_mirror_view_idx> (T)
+| |-------Grp_Interior
+| |  |-------Grp_Interior_Static
+| |  |-------Grp_Interior_Dynamic
+| |  | |-------Grp_Steering_Wheel (T)
+| |  | |-------Grp_Eyepoint (T)
+| |  | |-------Grp_Rearview_Mirror_Mounting <rearview_mirror_mounting_idx> (T)
+| |  | |-------Grp_Rearview_Mirror_View <rearview_mirror_view_idx> (T)
```
Comment (mettelilienthal) [inline content/geometry/vehicle-structure-model-structure.adoc:57]:
```suggestion

| |  | |-------Grp_Rearview_Mirror_View <rearview_mirror_view_idx> (T)
```

**Comment by ClemensLinnhoff** [comment]
Hm this does not really render as a list. At least not in my IDE. Could you fix that @mettelilienthal?

image

**Comment by mettelilienthal** [comment]
Which IDE are you using? Is the screenshot of the VSC preview function? If so, we are aware the preview does not render files as they will actually appear. That's why we suggested waiting until the pipeline is built so we can check how hierarchy levels will appear (and use PlantUML).
If you want to see the levels now already, we can change it to a simple (nested) list. However, we will need to change it again once the pipeline is done. We propose to keep the current version for now, to reduce efforts. Do you agree or would you prefer a simple list?

**Comment by ClemensLinnhoff** [comment]
> Which IDE are you using? Is the screenshot of the VSC preview function? If so, we are aware the preview does not render files as they will actually appear. That's why we suggested waiting until the pipeline is built so we can check how hierarchy levels will appear (and use PlantUML). If you want to see the levels now already, we can change it to a simple (nested) list. However, we will need to change it again once the pipeline is done. We propose to keep the current version for now, to reduce efforts. Do you agree or would you prefer a simple list?

Ah okay I understand. No, then let's leave it like this.

@LudwigFriedmann could you do a review? Or another possibility would be to just merge as is, see how it looks like in the online html document and then create new issues for potential fixes? I would actually prefer this to move forward.

**Pull request merged**